python/pyarrow/_dataset_parquet.pyx

```diff
@@ -711,6 +889,20 @@ cdef class ParquetFragmentScanOptions(FragmentScanOptions):
     cdef ArrowReaderProperties* arrow_reader_properties(self):
         return self.parquet_options.arrow_reader_properties.get()
 
+    IF PARQUET_ENCRYPTION_ENABLED:
+        @property
+        def parquet_decryption_config(self):
```

Comment:
We could also always add the property, but let it raise NotImplementedError when encryption is not enabled?

Reply:
> PARQUET_ENCRYPTION_ENABLED is a compile-time flag that would be set intentionally, having the code throw an exception at runtime could lead to confusion.

With the current code, a runtime exception would be raised anyway, it would just be a generic `AttributeError`. A specific exception could have a descriptive error message telling the user why the attributes are not available.

Reply:
> Do we want to define these functions with an ELSE statement and then throw an exception if they are called?

Yes, because right now the user would probably get a generic error such as `AttributeError("ParquetFragmentScanOptions has no attribute 'parquet_decryption_config'")`, which may also occur because of a typo in the attribute name, while it might be more informative to have something like `NotImplementedError("Trying access the 'parquet_decryption_config' attribute but encryption is not enabled in this build")`